src/coreclr/jit/objectalloc.cpp

```diff
@@ -747,27 +812,30 @@ void ObjectAllocator::ComputeStackObjectPointers(BitVecTraits* bitVecTraits)
                 // Check if this pointer always points to the stack.
                 // For OSR the reference may be pointing at the heap-allocated Tier0 version.
                 //
-                LclVarDsc* lclVarDsc = comp->lvaGetDesc(lclNum);
-
-                if ((lclVarDsc->lvSingleDef == 1) && !comp->opts.IsOSR())
+                if (!comp->opts.IsOSR())
```

Comment:
This is the heart of the change; the rest is there to ensure the deduction we make here is sound. 

We now need to know "everything" that can be assigned to each local.